sofarsounds/maestro
# use svg-url-loader to reduce build size

### svg-url-loader:

A webpack loader which loads SVG file as utf-8 encoded DataUrl string.

Existing url-loader always does Base64 encoding for data-uri. As SVG content is a human-readable xml string, using base64 encoding is not mandatory. Instead, one may only escape unsafe characters and replace " with ' as described in this article.

There are some benefits for choosing utf-8 encoding over base64.

Resulting string is shorter (can be ~2 times shorter for 2K-sized icons);
Resulting string will be compressed better when using gzip compression;
Browser parses utf-8 encoded string faster than its base64 equivalent.

### Description
- By using svg-url-loader the build size of maestro significantly reduces by ~45% (from 603 KB -> 348KB). 

### Motivation and Context
- Was done as one of the performance related tests.

### How Has This Been Tested?
- Run `yarn run analyze` on sofar-client and check the size of maestro buildpack when the browser loads.
- Then run the following commands on your local repos:
```cd $PROJECTS_DIR/maestro
yarn install
yarn build
yarn link
cd node_modules/react
yarn link
cd ../react-dom
yarn link
cd ../styled-component
yarn link
cd $PROJECTS_DIR/sofar-client
yarn link "@sofarsounds/maestro"
yarn link "react"
yarn link "react-dom"
yarn link "styled-components"
```
- Run `yarn run analyze` again on sofar-client and observe the size change in maestro buildpack. 

### Screenshots (if appropriate):
BEFORE:
![Screen Shot 2021-02-18 at 3 38 53 PM](https://user-images.githubusercontent.com/72412303/108418826-c2272a80-7257-11eb-9792-b3453c5bbb2e.png)
AFTER:
![Screen Shot 2021-02-18 at 3 38 42 PM](https://user-images.githubusercontent.com/72412303/108418851-c6ebde80-7257-11eb-863b-43f21a40ab08.png)


### Types of changes

- [x] Improvement (non-breaking change which improves functionality or refactors code)

### Checklist:
